iMovies/Feature/Dashboard/Data/Repository/RemoteUpcomingMoviesRepository.swift

```diff
@@ -15,24 +15,26 @@ final class RemoteUpcomingMoviesRepository: UpcomingMoviesRepository {
         self.httpClient = httpClient
     }
     
-    func fetchUpcomingMovies(completion: @escaping (Swift.Result<MovieDTO, Error>) -> Void) {
+    func fetchUpcomingMovies(completion: @escaping (Swift.Result<MovieDTO, APIError>) -> Void) {
         
-        let upcomingMoviesEndpoint = Endpoint<EndpointTypes.Private,MovieDTO>.upcoming
-        let request = upcomingMoviesEndpoint.makeRequest(with: AccessToken.accessToken)
+        let signedRequest = Endpoint.upcoming.makeRequest(with: AccessToken.accessToken)
         
-        httpClient.makeRequest(request: request) { result in
+        
+        httpClient.makeRequest(request: signedRequest) { result in
             switch result {
-            case .success(let (data,_)):
-                do {
-                    let decodedData = try JSONDecoder().decode(MovieDTO.self, from: data)
-                  
-                    completion(.success(decodedData))
-                } catch {
-                    print(error)
-                    completion(.failure(error))
+            case .success(let (data, response)):
+                let (movie,apiError) = MovieDTOMapper.map(data: data, response: response)
```

Comment:
Instead of mapping the API error here, I think we can check the status code in the HTTP client through the API error.